File: simp-api-ingredients/schemas/company.py
# schemas/company.py
from pydantic import BaseModel
import uuid
from typing import List

# For creating a company
class CompanyCreate(BaseModel):
    name: str

# For returning company data
class CompanyOut(BaseModel):
    company_id: uuid.UUID
    name: str

    class Config:
        from_attributes = True

# For paginated responses
class PaginatedCompanies(BaseModel):
    companies: List[CompanyOut]
    total: int

File: simp-front-end/src/lib/types/company.ts
// src/lib/types/company.ts

// Corresponds to CompanyOut schema
export type Company = {
  company_id: string; // UUID
  name: string;
};

// Corresponds to CompanyCreate schema
export type CompanyCreatePayload = {
  name: string;
};

// For paginated responses (matches PaginatedCompanies schema)
export type PaginatedCompanies = {
  companies: Company[];
  total: number;
};